great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1916,7 +1915,16 @@ def update_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
         Returns:
             The updated Checkpoint.
         """
-        return self.checkpoint_store.update_checkpoint(checkpoint)
+        result: Checkpoint | CheckpointConfig = self.checkpoint_store.update_checkpoint(
+            checkpoint
+        )
+        if isinstance(result, CheckpointConfig):
+            result = Checkpoint.instantiate_from_config_with_runtime_args(
+                checkpoint_config=result,
+                data_context=self,
+                name=Checkpoint.name,
```

Comment:
Should this be `Checkpoint`? The class itself? Or should it be `result.name`?

Reply:
Great catch! Fixing

Reply:
Addressed in https://github.com/great-expectations/great_expectations/pull/8084/commits/d21a474ecc16d15000eefc368c05fbee79ec5d63